### PR TITLE
[feature] Add objective/quest complete sound

### LINKED ISSUE #3526: Play sound when quest objective is finished
Play sound when quest objective is finished. So sleepy player can stop grinding quest mobs.

### PR DESCRIPTION
**Pull request by BreakBB**
## Issue references

Fixes #3526

## TODOs

- [x] Add settings to enable/disable both sounds
- [x] Add play button to test both sounds ???

## Proposed changes

- Play sound when an objective is complete
- Play a different sound when a quest is complete
- Don't play the objective complete sound, when it is the last objective of a quest

### CODE REVIEW

**Comment by drejjmit** [comment]
Should it really be on by default 🤔

**Comment by MyTechnoHunter** [comment]
agreed

**Comment by Dyaxler** [comment]
Agreed... off by default.

File: Modules/Options/QuestieOptionsDefaults.lua [hunk 1]
```diff
@@ -151,7 +151,9 @@ function QuestieOptionsDefaults:Load()
             showQuestsInNpcTooltip = true,
             trackerEnabled = true,
             ldbDisplayText = "Questie",
-            enableQuestFrameIcons = true
+            enableQuestFrameIcons = true,
+            soundOnQuestComplete = true,
+            soundOnObjectiveComplete = true,
```
Comment (BreakBB) [inline Modules/Options/QuestieOptionsDefaults.lua:156]:
```suggestion
            soundOnQuestComplete = false,
            soundOnObjectiveComplete = false,
```

File: Modules/Migration.lua [hunk 2]
```diff
@@ -204,6 +204,14 @@ local migrationFunctions = {
             Questie.db.global.currentHeaderEnabledSetting = false       -- set default
         end
     end,
+    [21] = function ()
+        if Questie.db.char.soundOnQuestComplete == nil then
+            Questie.db.char.soundOnQuestComplete = true
+        end
+        if Questie.db.char.soundOnObjectiveComplete == nil then
+            Questie.db.char.soundOnObjectiveComplete = true
+        end
```
Comment (BreakBB) [inline Modules/Migration.lua:213]:
```suggestion
        if Questie.db.char.soundOnQuestComplete == nil then -- new option
            Questie.db.char.soundOnQuestComplete = false
        end
        if Questie.db.char.soundOnObjectiveComplete == nil then -- new option
            Questie.db.char.soundOnObjectiveComplete = false
        end
```

**Pull request merged**